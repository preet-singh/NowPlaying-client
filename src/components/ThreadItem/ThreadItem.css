@media screen and (min-width: 320px){
    .thread-item {
    width: 95%;
    margin-left: auto;
    margin-right: auto;
    border: 1px solid gray;
    background-color: hsl(84, 60%, 45%);
    padding: 10px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    margin-bottom: 10px;
    position: relative;
    overflow: hidden;
  }

  .thread-item a {
    color: white;
    text-decoration: none;
    display: flex;
    flex-direction: column;
    height: 150px;
    width: 50%;
    margin-left: auto;
    margin-right: 25px;
  }

  .thread-item a:visited {
    color: white;
    text-decoration: none;
  }

  .thread-item a > img {
    position: absolute;
    height: 150px;
    width: 40%;
    object-fit: cover;
    min-height: 200px;
    left: 0px;
    top: 0px;
  }

  #play-button_home {
    position: absolute;
    bottom: -40px;
    right: 5px;
    width: 40px;
    height: 130px;
    border: none !important;
    border: none;
  }
}

@media screen and (min-width: 768px){
  .thread-item {
    width: 95%;
    height: 250px;
    margin-left: auto;
    margin-right: auto;
    border: 1px solid gray;
    background-color: hsl(84, 60%, 45%);
    padding: 10px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    margin-bottom: 10px;
    position: relative;
    overflow: hidden;
  }

  .thread-item:hover {
    background-color: hsl(84, 95%, 25%);
    cursor: pointer;
  }

  .thread-item a {
    /* background-color: lightcoral; */
    color: white;
    text-decoration: none;
    text-align: left;
    display: flex;
    flex-direction: column;
    height: 150px;
    padding: 10px;
    width: auto;
    margin-left: 40%;
    margin-right: auto;
  }

  .thread-item a h3 {
    font-size: 40px;
  }

  .thread-item a > p {
    font-size: 20px;
  }

  .thread-item a > img {
    position: absolute;
    width: 40%;
    height: auto;
    min-height: 300px;
    border-radius: initial;
    top: 0px;
    left: 0px;
    transform: translateY(-45px);
  }

  #play-button_home {
    position: absolute;
    top: 50px;
    right: 50px;
    width: 10%;
    height: 130px;
    border: none !important;
    border: none;
  }

  #last_comment {
    margin-top: 125px;
  }

}