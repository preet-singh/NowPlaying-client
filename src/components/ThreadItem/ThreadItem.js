import React from 'react';
import './ThreadItem.css';
import { Link } from 'react-router-dom';
import AuthService from '../../utils/auth-service';
import UserContext from '../../utils/context';
import moment from 'moment';
import playbutton from '../Images/play-button.svg'


class ThreadItem extends React.Component {
  static contextType = UserContext;
  constructor(props) {
    super(props);
    this.state = {
    }
  }
  

  getComments = async (context,id) => {
    let slicedWord = context.category.slice(0,context.category.length-1);
    if (slicedWord === 'movie') {
      AuthService.getMovieComments('movies',id)
      .then(comments => {
        if (comments.length > 0) {
          let final = comments[comments.length-1] || {}
          let date = moment(final.date_created).format('MMMM Do YYYY, h:mm:ss a')
          if (!this.state.lastComment) {
            this.setState({lastComment: date})
          }
        }
      })
    }
    else if (slicedWord === 'tv_show') {
      AuthService.getTVShowComments('tv_shows',id)
      .then(comments => {
        if (comments.length > 0) {
          let final = comments[comments.length-1] || {}
          let date = moment(final.date_created).format('MMMM Do YYYY, h:mm:ss a')
          if (!this.state.lastComment) {
            this.setState({lastComment: date})
          }
        }
      })
    }
    else if (slicedWord === 'podcast') {
      AuthService.getPodcastComments('podcasts',id)
      .then(comments => {
        if (comments.length > 0) {
          let final = comments[comments.length-1] || {}
          let date = moment(final.date_created).format('MMMM Do YYYY, h:mm:ss a')
          if (!this.state.lastComment) {
            this.setState({lastComment: date})
          }
        }
      })
    }
    else if (slicedWord === 'book') {
      AuthService.getBookComments('books',id)
      .then(comments => {
        if (comments.length > 0) {
          let final = comments[comments.length-1] || {}
          let date = moment(final.date_created).format('MMMM Do YYYY, h:mm:ss a')
          if (!this.state.lastComment) {
            this.setState({lastComment: date})
          }
        }
      })
    }
  }

  componentDidMount() {
    this.getComments(this.context,this.props.details.id)
  }
  render() {
    let lastCommentExists = `Last comment: ${this.state.lastComment}` || '';
    return (
      <div className='thread-item'>
        <Link to={`/${this.context.category}/${this.props.details.id}`}>
          <h3>{this.props.details.title}</h3>
          <img src='https://img01.mgo-images.com/image/thumbnail/v2/content/MMVAF76018A477C2826A4EC8747C40B7BE27.jpeg' alt='Movie poster'/>
          <p id='last_comment'>{this.state.lastComment? lastCommentExists : 'No comments yet! Be the first!'}</p>
        </Link>
        <img id='play-button' src={playbutton} alt='play button' />
      </div>
    )
  }
}

export default ThreadItem;