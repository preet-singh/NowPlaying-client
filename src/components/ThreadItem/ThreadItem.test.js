import React from 'react';
import ThreadItem from './ThreadItem.js';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`ThreadItem component`, () => {

  let details = {
      id: 20019,
      title: 'Avengers',
      poster: '2nvwe9d15fan3'
  }

  it('renders the ThreadItem', () => {
    const wrapper = shallow(<ThreadItem key={1} details={details}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})