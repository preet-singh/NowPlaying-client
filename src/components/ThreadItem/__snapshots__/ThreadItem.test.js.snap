// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadItem component renders the ThreadItem 1`] = `
<div
  className="thread-item"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <Link
    to="/undefined/20019"
  >
    <div
      className="thread-item-image-container"
    >
      <img
        alt="Movie backdrop"
        className="thread-item-poster"
        src="http://image.tmdb.org/t/p/w780/2nvwe9d15fan3"
      />
    </div>
    <div
      className="thread-item-info"
    >
      <h3>
        Avengers
      </h3>
      <p
        className="last-comment"
      >
        No comments yet! Be the first!
      </p>
      <div
        className="play-button-home-container"
        id="play-button-home-container"
      >
        <img
          alt="play button"
          className="playImage"
          src="white-play-button.png"
        />
      </div>
    </div>
  </Link>
</div>
`;
